### ISSUE TITLE
RFC: Parameters utility

### ISSUE THREAD

**Issue by saragerion** [aws-powertools/powertools-lambda-typescript#846]
```[tasklist]
### Tasks
- [ ] #1173
- [ ] #1175 
- [ ] #1176
- [ ] #1177 
- [ ] #1174 
- [ ] #1035 
- [ ] #1034 
- [ ] #1041 
- [ ] #1172 
- [ ] #1036 
- [ ] #1037 
- [ ] #1039
- [ ] #1042 
- [ ] #1043 
- [ ] #1040
```

## Key information

* RFC PR: (leave this empty)
* Related issue(s), if known: 
* * https://github.com/awslabs/aws-lambda-powertools-typescript/issues/484
* * https://github.com/awslabs/aws-lambda-powertools-typescript/issues/165
* Area: Parameters
* Meet [tenets](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/#tenets): Yes
* Approved by: ''
* Reviewed by: ''

## Summary
[summary]: #summary

Implementation/porting of a [Parameters](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/484) utility already implemented in Python. 

## Motivation
[motivation]: #motivation

Features parity across Powertools languages, prioritized based on community demand.

I am keeping the sections below empty for now to allow us to think it throught, but I am keen on hearing our community's ideas, wishlists, desired UX, etc.

## Proposal
[proposal]: #proposal

## Drawbacks
[drawbacks]: #drawbacks

## Rationale and alternatives
[rationale-and-alternatives]: #rationale-and-alternatives

## Unresolved questions
[unresolved-questions]: #unresolved-questions

**Comment by dreamorosi** [comment]
Just for info purposes, there is a development branch that I started about 10 months ago with an initial implementation of the utility that matches the Python version of the time, plus unit tests.

As I mentioned I haven't been working on this for a long while, I have just rebased today to incorporate the current project structure. Maybe this can be used as a PoC reference for a future implementation. In case we want to go in a different direction I'm also happy to discard it.

**Comment by dreamorosi** [comment]
The first public beta for the utility has been released in `v1.7.0` (`@aws-lambda-powertools/parameters@1.7.0-beta` on npm).

We are eager to hear your feedback, so feel free to open an issue if you encounter any bug or open a discussion here or on Discord if you have questions or doubts.
